### DISCUSSION TITLE
Roadmap & Goals

### DISCUSSION THREAD

**Discussion by killix** [OpenFunction/OpenFunction#357] [Q&A]
Hello team,

I'm evaluating the possibility of using OpenFunction for a new open-source project. Currently, my POC works with Openfaas.
The ideal would be to switch to OpenFunction, however, I need some features first. Like support for Rust and Docker images for functions.
Do you have goals for the end of the year? If so can you share them, to give better visibility on your calendar?

Thanks

**Comment by benjaminhuo** [comment]
@killix Thanks for the attention. Do you mean you need OpenFunction to support Rust functions in the same way like Go functions now?

**Reply by killix** [reply]
At least yes, and also ideally the [Support building functions with Dockerfile](https://github.com/OpenFunction/OpenFunction/issues/204). Sorry if i was not clear.

**Reply by benjaminhuo** [reply]
Got you. Rust functions with the Dockerfile build method. It is possible to support that by the end of this year. 
In which project do you want to use OpenFunction?
BTW we also welcome contributors to get involved.

**Reply by killix** [reply]
For now, I am building an early experimental PoC, to see if it's worth continuing. The idea is to create an alternative to Supabase with some others keys decisions.

**Comment by nadworny** [comment]
hi guys, I'm not sure if I should ask here or on dapr but do you also plan to support dapr for python & java? If I understand this [case study](https://blog.dapr.io/posts/2022/05/14/how-dapr-helps-to-build-a-cloud-agnostic-faas-platform/#the-functions-framework) correctly, it's not supported yet.

**Reply by benjaminhuo** [reply]
> hi guys, I'm not sure if I should ask here or on dapr but do you also plan to support dapr for python & java? If I understand this [case study](https://blog.dapr.io/posts/2022/05/14/how-dapr-helps-to-build-a-cloud-agnostic-faas-platform/#the-functions-framework) correctly, it's not supported yet.

@nadworny , OpenFunction already supports Dapr in the Go, Nodejs, and Java functions-framework. Python is not yet supported.

You can find Go, Nodejs, and Java function samples with Dapr support in the following links:

https://openfunction.dev/docs/getting-started/quickstarts/sync-functions/
https://openfunction.dev/docs/getting-started/quickstarts/async-functions/

**Reply by nadworny** [reply]
awesome, thanks a lot @benjaminhuo. Are you planning to support python as well? Is there a feature/issue that I could follow for that?

**Reply by benjaminhuo** [reply]
> awesome, thanks a lot @benjaminhuo. Are you planning to support python as well? Is there a feature/issue that I could follow for that?

Here: https://github.com/OpenFunction/functions-framework/issues/18
